Add documentation about MailChannels domain lockdown (#12057)
cloudflare/cloudflare-docs: content/pages/platform/functions/plugins/mailchannels.md

{{<Aside type="note">}}
To use the Mailchannels Pages Plugin, you must first add a [Domain Lockdown DNS record](https://support.mailchannels.com/hc/en-us/articles/16918954360845-Secure-your-domain-name-against-spoofing-with-Domain-Lockdown-) to the domain you are sending email from. For more information, refer to the [Mailchannels documentation](https://support.mailchannels.com/hc/en-us/articles/4565898358413-Sending-Email-from-Cloudflare-Workers-using-MailChannels-Send-API).

For standard setup with Cloudflare Pages, see [Enabling MailChannels for your Account](#enabling-mailchannels-for-your-account---domain-lockdown) below.

> {{</Aside>}}

Reviewer: ```suggestion

{{</Aside>}}
```

For more information about MailChannels and the options they support, refer to [the documentation](https://mailchannels.zendesk.com/hc/en-us/articles/4565898358413-Sending-Email-from-Cloudflare-Workers-using-MailChannels-Send-API).

## Enabling MailChannels for your Account - Domain Lockdown

To enable MailChannels to send email on your behalf through Cloudflare Pages Functions, you have to setup a [Domain Lockdown DNS record](https://support.mailchannels.com/hc/en-us/articles/16918954360845-Secure-your-domain-name-against-spoofing-with-Domain-Lockdown-) for the domain you are sending email from.

For most cases, all this involves is adding a `TXT` entry to your DNS.

> First, grab your account's workers subdomain:

Reviewer 2: ```suggestion
First, find your account's `workers.dev` subdomain:
```

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.

> 2. In Account Home, select **Workers & Pages**, then **Overview**.

Reviewer 2: ```suggestion
2. Select **Workers & Pages** > **Overview**.
```

> 3. On the right-hand side, note your worker's *Subdomain*. It's likely something similar to `myaccount.workers.dev`.

Reviewer 2: ```suggestion
3. On the right-hand side of **Overview**, note your `workers.dev` **Subdomain**. It will be similar to `myaccount.workers.dev` and customizable by you.
```

> Then, add the MailChannels DNS record:

Reviewer 2: ```suggestion
After you have found your `workers.dev` subdomain, add the MailChannels DNS record:
```

1. In **Account Home**, select the website you would like to add an SPF record for.
2. Select **DNS** > **Records** > **Add Record**.

> 3. Add the following TXT DNS record, replacing `myaccount.workers.dev` with your own workers subdomain.

Reviewer 2: ```suggestion
3. Add the following TXT DNS record, replacing `myaccount.workers.dev` with your own `workers.dev` subdomain.
```

{{<example>}}

| Type | Name            | Content                            |
|------|-----------------|------------------------------------|
| TXT  | `_mailchannels` | `v=mc1 cfid=myaccount.workers.dev` |

{{</example>}}

> You can find more details about the domain lockdown record and more complex use-cases in MailChannel's [Domain Lockdown DNS record](https://support.mailchannels.com/hc/en-us/articles/16918954360845-Secure-your-domain-name-against-spoofing-with-Domain-Lockdown-) support article.

Reviewer 2: ```suggestion
Find more details about the domain lockdown record and more complex use cases in MailChannel's [Domain Lockdown DNS record](https://support.mailchannels.com/hc/en-us/articles/16918954360845-Secure-your-domain-name-against-spoofing-with-Domain-Lockdown-) support article.
```

## SPF support for MailChannels

To use both MailChannels and Cloudflare Email Routing:

1. In **Account Home**, select the website you would like to add an SPF record for.
2. Select **DNS** > **Records** > **Add Record**.
3. Add the following TXT DNS record:

{{<example>}}

| Type | Name | Content                                                                     |
|------|------|-----------------------------------------------------------------------------|
| TXT  | `@`  | `v=spf1 include:_spf.mx.cloudflare.net include:relay.mailchannels.net -all` |

{{</example>}}

> {{<Aside type= "note" header="Existing SPF Record">}}

Reviewer 2: ```suggestion
{{<Aside type= "note" header="Existing SPF record">}}
```

>  If you have an existing SPF record for your domain, add the `include:...` entries to it instead of creating a new one.

Reviewer 2: ```suggestion
If you have an existing SPF record for your domain, add the `include:...` entries to it instead of creating a new one.
```

To add a DKIM signature to a message, add the following fields to the `personalization` object for the message:

> **`dkim_domain`**: The domain you are sending the email from. For example, if you are sending an email from `support@mydomin.com`, set this to `mydomain.com`.

Reviewer 2: ```suggestion
**`dkim_domain`**: The domain you are sending the email from. For example, if you are sending an email from `support@mydomain.com`, set this to `mydomain.com`.
```

3. Create a DNS record with the content of the generated `dkim_record.txt` file content, following the steps below.

Next, look in your generated `dkim_record.txt` file for your DKIM credentials, and add them to your website in the Cloudflare dashboard. Follow the steps below:

1. In Account Home, select the website you would like to add a DKIM record.
2. In the menu on the left select **DNS** > **Records** > **Add Record**.
3. Add the following TXT DNS record:

{{<example>}}

| Type | Name                      | Content                                  |
|------|---------------------------|------------------------------------------|
| TXT  | `mailchannels._domainkey` | Insert the contents of `dkim_record.txt` | 

{{</example>}}

> {{<Aside type= "note" header="Use a Different Selector">}}

Reviewer 2: ```suggestion
{{<Aside type= "note" header="Use a different selector">}}
```

> You can substitute `mailchannels.domainkey` for any name of the format `<selector key>._domainkey`. You can choose any value as the selector, as long as it is permitted as a DNS hostname (that is, all lowercase letters, numbers and hyphens).

Reviewer 2: ```suggestion
You can substitute `mailchannels.domainkey` for any name with the format `<selector_key>._domainkey`. You can choose any value as the selector, as long as it is permitted as a DNS hostname (that is, all lowercase letters, numbers and hyphens).
```
